perceval/runtime/payload_generator.py

```diff
@@ -0,0 +1,97 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import uuid
+
+from perceval.components import Experiment
+from perceval.serialization import serialize
+from perceval.utils import PMetadata
+
+__process_id__ = uuid.uuid4()
+
+
+
+class PayloadGenerator:
+
+    @staticmethod
+    def generate_payload(command: str,
+                         experiment: Experiment = None,
+                         params: dict[str, any] = None,
+                         platform_name: str = "",
+                         **kwargs
+                         ) -> dict[str, any]:
+        r"""
+        Generate a simple payload containing the experiment, with the following template:
+        {
+            'pcvl_version': ...
+            'process_id': ...
+            'platform_name': ...
+            'payload': {
+                'command': ...
+                'kwarg1': ...
+                'kwarg2': ...
+                ...
+                'parameters': ...
+                'experiment': ...
+            }
+        }
+
+        :param command: name of the method used
+        :param experiment: (optional) Perceval experiment, by default an empty Experiment will be generated
+        :param params: (optional) parameters to be listed in the 'parameters' section of the payload
+        :param platform_name: (optional) name of the platform used
+
+        Other parameters can be added to the payload via **kwargs.
+        """
+
+        if experiment is None:
+            experiment = Experiment()
+
+        j = {
+            'pcvl_version': PMetadata.short_version(),
+            'process_id': str(__process_id__)
+        }
+        if platform_name:
+            j['platform_name'] = platform_name
+
+        payload = {
+            'command': command,
+            **kwargs
+        }
+
+        if experiment.input_state:
+            experiment.check_input(experiment.remove_heralded_modes(experiment.input_state))
```

Comment:
`check_input` is redefined by RemoteProcessor. The check using `experiment` has already been done when using `with_input`. 
The best is to put it in `RemoteProcessor.prepare_job_payload`